### ISSUE TITLE
Drag other components onto gridster

### ISSUE THREAD

**Issue by Pro** [tiberiuzuld/angular-gridster2#357]
I have two main components on my page, side by side:
- 50% width is a tree view (unrelated to gridster)
- 50% width is a view component with gridster

My plan is now to take a node from the tree view and drag it onto the gridster. At dragEnter a new `GridsterItem` should be created and it's position should be forced to the current mouse position (i.e. all the other grid items should be pushed away). When moving around the mouse, the grid item should also be moved (similar as taking an already existing grid item and dragging it around).

Unfortunately I'm not able to force a newly added item to the current mouse position, it is always added somewhere at the end because there is a collission. Additionally the grid item is not moved around.

I already tried the following:

On my view component (which contains the gridster)  I have an event listener for the `dragEnter()`:

```typescript
	public onDragEnter(event: DragEvent): void {
		if (this.dragInsertGridItem != null)
			return; // we are already dragging

		let itm = {
			x: this.gridster.pixelsToPositionX(event.layerX, Math.ceil),
			y: this.gridster.pixelsToPositionY(event.layerY, Math.ceil),
			cols: 1,
			rows: 1,
			label: "Some label",
			initCallback: (item, itemComponent) => {
				this.dragInsertGridItem = itm;
				this.dragInsertGridItemComponent = itemComponent;
				this.dragInsertGridItemComponent.drag.dragStart(event);
			}
		};
		this.dashboard.push(itm);

	}
```

It adds the new item to the dashboard and sets some local variables. 

The drag over looks like this:
```typescript
	public onDragOver(event: DragEvent): boolean {
		if (event.preventDefault) {
			event.preventDefault(); // Necessary. Allows us to drop.
		}
		//console.log("Drag over", event);
		if (this.dragInsertGridItem == null)
			return false;

		this.dragInsertGridItemComponent.drag.dragMove(event);
		return false;
	}
```

The dragged item just jumps around and is flickering.

What's the right way to achieve this behaviour, that I'm creating a new grid item with the API and then moving it around through the API?

**Comment by saiar** [comment]
Hi @Pro , were you able to figure out the solution to this? I am also looking to implement this functionality of being able to drag gridItems from one gridster container and drop them into another gridster container but haven't figured it out yet.

**Comment by Pro** [comment]
No, unfortunately I didn't find a solution yet. For my use-case it is not specifically a gridster, where I grab the item, it is more general any kind of item from another component which is dragged onto the gridster.

@tiberiuzuld could you post some hints how this could be achieved? That would be very helpful!

**Comment by saiar** [comment]
@Pro Thanks for responding.

Ah, actually that was/is my case as well. But if it makes it any easier to get this functionality working if the source is a gridster container, I'd be ready to make that change.

Yep, would be very helpful to have any hints to solve this.

**Comment by anji405** [comment]
@Pro 
I have similar application to introduce new gridster items via drag from other draggable container panel.
So, can you let me know if you have good move implementing that..

**Comment by Pro** [comment]
Nope, unfortunately I am still missing a working implementation

**Comment by anji405** [comment]
So, do we need to shift to angular 2gridster then..
Please assist with your thoughts on this

**Comment by Konstruktour** [comment]
Related to [Preview of element when added by drag&drop #254](https://github.com/tiberiuzuld/angular-gridster2/issues/254 )
would appreciate if gridster would support this kind of feature
